airstack-modules/modules/airstack/domain-name/domain-name.ts

```diff
@@ -0,0 +1,1152 @@
+    let domain = getOrCreateAirDomain(new Domain(node, chainId, block, tokenAddress));
+    if (fromOldRegistry && domain.isMigrated == true) {
+      // this domain was migrated from the old registry, so we don't need to hanlde old registry event now
+      return;
+    }
+    let previousOwnerId = domain.owner;
+    if (previousOwnerId == null) {
+      previousOwnerId = getOrCreateAirAccount(chainId, ZERO_ADDRESS).id;
+    }
+    let id = createEntityId(transactionHash, blockHeight, logIndex);
+    let entity = AirDomainTransferTransaction.load(id);
+    if (entity == null) {
+      entity = new AirDomainTransferTransaction(id);
+      entity.from = previousOwnerId;
+      entity.to = getOrCreateAirAccount(chainId, newOwnerAddress).id;
+      let airBlock = getOrCreateAirBlock(
+        chainId,
+        blockHeight,
+        blockHash,
+        blockTimestamp,
+      );
+      entity.block = airBlock.id;
+      entity.transactionHash = transactionHash;
+      entity.tokenId = domain.tokenId;
+      entity.domain = domain.id;
+      entity.index = updateAirEntityCounter(AIR_DOMAIN_TRANSFER_ENTITY_COUNTER_ID, airBlock);
+      entity.save();
+    }
+  }
+
+  /**
+   * @dev This function tracks a new resolver transaction
+   * @param resolver resolver contract address
+   * @param node domain node
+   * @param chainId chain id
+   * @param blockHeight block number in the chain
+   * @param blockHash block hash
+   * @param blockTimestamp block timestamp
+   * @param transactionHash transaction hash
+   * @param logIndex event log index
+   * @param tokenAddress contract address of nft token
+   * @param fromOldRegistry specifies if the event is from the old registry
+   */
+  export function trackDomainNewResolverTransaction(
+    resolver: string,
+    node: string,
+    chainId: string,
+    blockHeight: BigInt,
+    blockHash: string,
+    blockTimestamp: BigInt,
+    transactionHash: string,
+    logIndex: BigInt,
+    tokenAddress: string,
+    fromOldRegistry: boolean,
+  ): void {
+    // get block
+    let block = getOrCreateAirBlock(chainId, blockHeight, blockHash, blockTimestamp);
+    // get domain
+    let domain = getOrCreateAirDomain(new Domain(node, chainId, block, tokenAddress));
+
+    if (fromOldRegistry && node != ROOT_NODE && domain.isMigrated == true) {
+      // this domain was migrated from the old registry, so we don't need to hanlde old registry event now
+      return;
+    }
+    // get previous resolver
+    let previousResolverId = domain.resolver;
+    // create new resolver
+    let resolverEntity = getOrCreateAirResolver(domain, chainId, resolver);
+    // update domain resolver
+    domain.resolver = resolverEntity.id;
+    // update domain resolved address
+    if (resolverEntity.resolvedAddress) {
+      domain.resolvedAddress = resolverEntity.resolvedAddress;
+    }
+    // do recursive domain delete
+    domain.lastBlock = block.id;
+    recurseSubdomainCountDecrement(domain, chainId, block, tokenAddress);
```

Comment:
this is because when you change the resolver for a domain, your subdomains are no longer linked to the same resolver.
But, I would like to investigate more on this.

Reply:
Found this issue from the ens subgraph GitHub repo - https://github.com/ensdomains/ens-subgraph/pull/44